src/modeler/AutoRest.Swagger/Validation/TrackedResourcePatchOperationValidation.cs

```diff
@@ -46,8 +46,8 @@ public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<
             IEnumerable<Operation> patchOperations = ValidationUtilities.GetOperationsByRequestMethod("patch", serviceDefinition);
 
             // enumerate all the models returned by all PATCH operations (200/201 responses)
-            var respModels = patchOperations.Select(op => op.Responses["200"]?.Schema?.Reference?.StripDefinitionPath());
-            respModels.Union(patchOperations.Select(op => op.Responses["201"]?.Schema?.Reference?.StripDefinitionPath())).Where(modelName=>!string.IsNullOrEmpty(modelName));
+            var respModels = patchOperations.Select(op => op.Responses?["200"]?.Schema?.Reference?.StripDefinitionPath());
+            respModels.Union(patchOperations.Select(op => op.Responses?["201"]?.Schema?.Reference?.StripDefinitionPath())).Where(modelName=>!string.IsNullOrEmpty(modelName));
```

Comment:
Just ran it against an actual swagger spec... should have failed for most specs in the repo. As always: Would be best to run rules against the entire repo.